docs/docs/developers/tutorials/codealong/first_fees.md

```diff
@@ -0,0 +1,354 @@
+
+### Tools
+
+**TODO: as snippet**
+To use Aztec's suite of tools you'll need to:
+ - [Get docker](https://docs.docker.com/get-started/get-docker/)
+ - Run `bash -i <(curl -s https://install.aztec.network)`
+
+Test the cli wallet with: `aztec-wallet --version`
+
+By default the sandbox runs everything including a pxe. For this tutorial, and more realistically when using testnet, we will be using PXEs client side. Eg as part of a wallet.
+
+Start the sandbox (L1, L2, but not the PXE) via: `NO_PXE=true aztec start --sandbox`
+
+### Specifying the network URL in commands
+
+When using `aztec-wallet`, the Aztec network node to connect to can be specified in each relevant command:
+
+```bash
+aztec-wallet --node-url <string> ...
+```
+
+The string currently defaults to the the local sandbox: `http://host.docker.internal:8080`, or can be a bootnode url.
+More conveniently, this can be specified via the environment variable: `AZTEC_NODE_URL`.
+
+The equivalent using aztec.js - create a PXE pointing to the desired url:
+
+```javascript
+import { createPXEClient, waitForPXE, PXE } from "@aztec/aztec.js";
+async function main() {
+  const { PXE_URL = 'http://localhost:8080' } = process.env; // from env var
+  const pxe = await createPXEClient(PXE_URL);
+  await waitForPXE(pxe);
+  // use pxe...
+}
+```
+
+## Create Account Contract in a PXE
+
+**TODO: as snippet**
+
+An account on Aztec is a smart contract that specifies a method of authentication and a method of payment, allowing it to be used by the protocol to perform a transaction.
+
+For convenience, Aztec has implemented an account contract that authenticates transactions using Schnorr signatures. The contract class for a Schnorr account is pre-registered on the network to bootstrap first use. Ordinarily for a contract to be deployed, its class would have to be register with the network first.
+
+When a PXE creates a Schnorr account, there are three key things that occur:
+ - generation of keys privately
+ - registeration of a Schnorr account locally (effectively an instance of the Schnorr account class)
+   - at this stage the address of the account is known, even though it is not yet deployed
+ - initialization and deployment of the account contract to the network
+
+**Create the keys and account in the PXE**
+
+Lets first create the account but only register the instance in the pxe. Its address will be calculated, and in a later step we can deploy it counterfactually to the network.
+
+```bash
+aztec-wallet create-account --register-only -a main
+```
+
+The `-a main` sets this new account's alias to, "main". Use `aztec-wallet get-alias` (with no params) to see all aliases.
+
+The equivalent using aztec.js - create a random account locally (we will deploy it in the next step):
+
+```javascript
+import { Fr } from "@aztec/aztec.js";
+import { getSchnorrAccount } from '@aztec/accounts/schnorr';
+async function main() {
+  let secretKey = Fr.random(); let salt = Fr.random();
```

Comment:
It's not clear in this snippet that the code from the above snippet is required for this to work

Reply:
Added a code comment